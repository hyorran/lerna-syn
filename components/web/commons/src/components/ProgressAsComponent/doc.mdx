---
name: ProgressAsComponent
menu: Commons
---

import { Playground, PropsTable } from 'docz'
import ProgressAsComponent, { ComponentWithProps } from './'

# ProgressAsComponent

A Simple ProgressAsComponent Example 

## Basic Usage
<Playground>
  <div>
    <ProgressAsComponent />
  </div>
</Playground>

## Props

<PropsTable of={ ComponentWithProps } />
