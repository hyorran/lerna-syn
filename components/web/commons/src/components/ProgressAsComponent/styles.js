// import 'react-sweet-progress/lib/style.css'

export default () => ({

})
