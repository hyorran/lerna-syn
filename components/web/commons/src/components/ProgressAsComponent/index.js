import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import { Progress } from 'react-sweet-progress'
import Colors from '@syntesis/c-styles/src/styles/Colors'

import styles from './styles'

const ProgressAsComponent = (props) => {
  const {
    value
  } = props

  let status = ''

  if (value <= 25) {
    status = 'red'
  } else if (value > 25 && value <= 75) {
    status = 'orange'
  } else if (value > 75) {
    status = 'green'
  }

  return (
    <Progress
      percent={ value }
      theme={ {
        red: {
          color: Colors.error.main
        },
        orange: {
          color: Colors.warning.main
        },
        green: {
          color: Colors.success.main
        }
      } }
      status={ status }
    />
  )
}

ProgressAsComponent.propTypes = {
  /** -*/
  value: PropTypes.number
};

ProgressAsComponent.defaultProps = {
  value: 0
};

export default withStyles(styles)(ProgressAsComponent)
export { ProgressAsComponent as ComponentWithProps }
