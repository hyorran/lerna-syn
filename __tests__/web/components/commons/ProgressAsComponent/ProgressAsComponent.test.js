import React from 'react'
import ProgressAsComponent from '@syntesis/c-commons/src/components/ProgressAsComponent'
import { mount } from 'enzyme'

describe('<ProgressAsComponent />', () => {
  it('rendered', () => {
    const wrapper = mount((
      <ProgressAsComponent value={ 50 } />
    ))
    expect(wrapper).toMatchSnapshot()
  })
})
