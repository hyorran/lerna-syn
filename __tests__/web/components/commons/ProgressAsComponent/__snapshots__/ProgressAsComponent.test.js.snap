// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressAsComponent /> rendered 1`] = `
<WithStyles(ProgressAsComponent)
  value={50}
>
  <ProgressAsComponent
    classes={Object {}}
    value={50}
  >
    <t
      percent={50}
      status="orange"
      theme={
        Object {
          "green": Object {
            "color": "#4caf50",
          },
          "orange": Object {
            "color": "#ffa000",
          },
          "red": Object {
            "color": "#d32f2f",
          },
        }
      }
    >
      <div
        className="react-sweet-progress"
      >
        <s
          background="#ffa000"
          percent={50}
          prefixClass="react-sweet-progress"
          status="orange"
        >
          <div
            className="react-sweet-progress-line"
            style={
              Object {
                "backgroundColor": undefined,
              }
            }
          >
            <div
              className="react-sweet-progress-line-inner undefined"
              style={
                Object {
                  "backgroundColor": "#ffa000",
                  "width": "50%",
                }
              }
            />
          </div>
        </s>
        <div
          className="react-sweet-progress-symbol"
        >
          50%
        </div>
      </div>
    </t>
  </ProgressAsComponent>
</WithStyles(ProgressAsComponent)>
`;
